ddtrace/tracer/tracer.go

```diff
@@ -450,6 +453,33 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 		// if not already sampled or a brand new trace, sample it
 		t.sample(span)
 	}
+	var labels []string
+	if t.config.profilerHotspots {
+		labels = append(labels, "span id", fmt.Sprintf("%d", span.SpanID))
+	}
+	// nil checks might not be needed, but better be safe than sorry
+	if span.context.trace != nil && span.context.trace.root != nil {
+		localRootSpan := span.context.trace.root
+		if t.config.profilerHotspots {
+			labels = append(labels, "local root span id", fmt.Sprintf("%d", localRootSpan.SpanID))
+		}
+		if t.config.profilerEndpoints {
+			// TODO(REVIEWERS) this MUST NOT contain personally identifiable
+			// information, is it safe to assume that this guarantee will be met
+			// here?
```

Comment:
I don't think it's safe to make such an assumption. It's very likely that it can. It could be a URL, or it could be an SQL query that is un-obfuscated (the agent does this, but we should start doing it in the client at some point).

Reply:
Ok. But SQL queries should usually not be local root spans, right? Anyway, I agree that there are edge cases where PII could leak here. I'll discuss this internally to see how strict we have to be on that.

Reply:
If someone has a program where all they do is SQL queries, they very well may be. I think that shouldn't be that uncommon of a use case.

Reply:
@gbbr okay had some more discussions within the team, and I'll change the code to check if the resource type is http, will ping you when that's done.

Reply:
@gbbr 34948c73cc718d61e62f297e0b9883c036a2a96b changes the logic so that endpoints (resources) are only applied for "web" requests. That should prevent SQL queries and stuff from getting labels, right?